### PR TITLE
Refactored ItineraryBuilder, connected UI of the tags part of the itinerary creation process

### PR DESCRIPTION
**Pull request by YofiY**
- refactored ItineraryBuilder:	
        - renamed `visible` attribute to `isPublic`
	- changed `time`'s type from `Int` to `Float`

- Connected the UI to the backend inside of `CreationStepChooseTags.kt`

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/DemoSetup.kt [hunk 1]
```diff
@@ -71,7 +71,7 @@ fun addItineraries(
           title = "Shopping then adventure",
           tags = listOf(ItineraryTags.ADVENTURE, ItineraryTags.LUXURY),
           description = "gucci",
-          visible = true,
+          isPublic = true,
```
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/DemoSetup.kt:74]: name is clearer

File: app/src/main/java/com/github/wanderwise_inc/app/DemoSetup.kt [hunk 2]
```diff
@@ -90,8 +90,8 @@ fun addItineraries(
                   ItineraryTags.WILDLIFE,
                   ItineraryTags.WELLNESS),
           description = null,
-          visible = true,
-          time = 3,
+          isPublic = true,
+          time = 3f,
```
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/DemoSetup.kt:94]: Time was initially a float before, but we changed it to an integer as we felt that it was overkill to have values like "1.3" hours.
We can change it, but I think we should chat and see if we agree on the extra precision

**Review by ethangraham2001 — COMMENTED** [review]
Looks good but there are a couple additional considerations

- the offline storage stack needs to be modified, notably the protobuf files and serializers
- The firestore data that we have currently will need to be modified

This introduces more work that someone is going to have to do, albeit not a huge amount. Nonetheless, we should probably discuss during the next standup meeting.

**Review by gunnm0 — CHANGES_REQUESTED** [review]
There are a few issues I see with this code, firstly if we change to the next screen since you don't compose the values of CreateItineraryViewModel.newItineraryBuilder, nothing gets composed and you can't see the value that were previously written. Also you should at least reach 80% test coverage before the code is pushed into main.

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt [hunk 3]
```diff
@@ -83,4 +103,15 @@ class CreateItineraryViewModel(
     // return the notSetValues (empty if everything has been set)
     return notSetValues
   }
+
+  fun uploadItineraryBannerPictureToStorage(userUid: String) {
+    imageRepository.uploadImageToStorage("profilePicture/${userUid}")
```
Comment (gunnm0) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt:130]: First of all, why is the itineraryBannerPicture saved at the path profilePicture/${userUid}, this task is more tricky than this, we need to push the image with the itinerary uid which you can only get once the intinerary has been pushed to firebase.

**Review by gunnm0 — CHANGES_REQUESTED** [review]
there are issues with this code as well as missing tests.

**Pull request closed without merging**